src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -90,10 +95,66 @@ public override void Open()
             var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE;
             flags |= (ConnectionStringBuilder.Cache == SqliteConnectionCacheMode.Shared) ? Constants.SQLITE_OPEN_SHAREDCACHE : Constants.SQLITE_OPEN_PRIVATECACHE;
 
-            var rc = NativeMethods.sqlite3_open_v2(ConnectionStringBuilder.DataSource, out _db, flags, vfs: null);
+            var path = AdjustForRelativeDirectory(ConnectionStringBuilder.DataSource);
+
+            var rc = NativeMethods.sqlite3_open_v2(path, out _db, flags, vfs: null);
             MarshalEx.ThrowExceptionForRC(rc, _db);
             SetState(ConnectionState.Open);
+
+            SetFolders();
+        }
+
+#if !NETCORE50
+        private void SetFolders() { }
+
+        private string AdjustForRelativeDirectory(string path)
+            => path;
+#else
+        private string AdjustForRelativeDirectory(string path)
+        {
+            var appData = GetApplicationData();
+            try
+            {
+                if (appData == null || appData?.LocalFolder?.Path == null || Path.IsPathRooted(path))
+                {
+                    return path;
+                }
+
+                return Path.GetFullPath(Path.Combine(appData.LocalFolder.Path, path));
+            }
+            catch (NotSupportedException)
```

Comment:
GetFullPath. If a user tried to specify a valid [URI filename](https://www.sqlite.org/c3ref/open.html#urifilenameexamples), e.g. `file://( localfolder )/data.db`